x/mongo/driver/topology/topology.go

```diff
@@ -338,6 +396,11 @@ func (t *Topology) Connect() error {
 		}
 		go t.pollSRVRecords(uri.Host)
 		t.pollingwg.Add(1)
+	} else {
+		logger := newHostLogger(t.cfg.logger)
+		for _, host := range parsedHosts {
+			logger.log(host)
```

Comment:
Before we log, we first need to check if it is appropriate to log. We should update `mustLogTopologyMessage` to the following:

```go
func mustLogTopologyMessage(topo *Topology, level logger.Level) bool {
	return topo.cfg.logger != nil && topo.cfg.logger.LevelComponentEnabled(
		level, logger.ComponentTopology)
}
```

and then wrap this block:

```go
		if mustLogTopologyMessage(t, logger.LevelInfo) {
			hostSet := map[string]bool{}
			for _, host := range parsedHosts {
				if h, _, err := net.SplitHostPort(host); err == nil {
					host = h
				}

				if !hostSet[host] {
					logTopologyWrongServer(t, host)
				}

				hostSet[host] = true
			}
		}
```